docs/modules/ROOT/pages/adr/0039-servala-framework-requirements.adoc

```diff
@@ -0,0 +1,142 @@
+* be able to map the Helm Chart values to the API values for
+** backend variables, for example backup, maintenance.
+** user-facing variables, for example instances, settings.
+* be able to manipulate API for user-facing variables.
+
+=== Service Operator
+
+NOTE: The service operator takes care of the individual service instances.
+
+_As a service operator, I want to_:
+
+* get informed when a service instance behaves different than its defined and agreed good state
+* know what to do when a service acts outside its defined good state
+* rely on tooling which helps to troubleshoot potentially misbehaving service instances
+* be able to handle support requests by having useful information about service instance at hand
+
+=== Framework Engineer
+
+NOTE: The framework engineer develops the framework tooling and the tools required to make day-2 operations possible.
+
+_As a framework engineer, I want to_:
+
+* provide the <<Service Maintainer>> with an easy to use framework for them to build managed services
+* build a rock-solid base which others can rely on
+* be able to easily develop the framework
+* have an extensive testing infrastructure in place
+* ensure backward compatibility when updating framework
+* provide migration guides for breaking changes
+* deprecate old features gracefully
+
+=== Platform Operator
+
+NOTE: The platform operator takes care of the underlying platform where the services are running on.
+
+_As a platform operator, I want to_:
+
+* have clear separation of concerns between the services and the platform
+* have strong multi-tenancy to protect different tenants using the platform
+* protect service instances from each other (noisy neighbours)
+
+=== Self-Service Portal
+
+NOTE: The self-service portal consumes the catalog of managed services built by using this framework.
+
+_As a self-service portal, I want to_:
+
+* query a well-defined API to get a list of available services and their metadata
+* retrieve the available parameters of a service to provide the user a configuration form
+* be sure that the API is standardized so that I can rely on it
+
+== Core Requirements
+
+Kubernetes API::
+The framework and the control-plane is based on the Kubernetes API and its tooling.
+This is the core technology used at VSHN and has the most experience around.
+Moving away from the Kubernetes API as the core architecture would be a significant change with plenty of risks involved.
+
+Core Tooling::
+The framework relies on many tools, assuming these tools are available and ready to use.
+This core tooling must be industry standard whenever possible, so that compatibility is given.
+Examples: Prometheus for metrics collection.
+
+Independence::
+The framework must be independent of VSHN Managed Kubernetes and VSHN Managed OpenShift.
+This mainly means it must not depend on Project Syn to allow deploying the service catalog to non-VSHN managed Kubernetes environments.
```

Comment:
Does this also mean we should not depend on specific S3 implementations? That is usually the other important API we need for almost every service.